planner: change default of tidb_opt_prefer_range_scan (#18876)
pingcap/docs: system-variables.md

### tidb_opt_prefer_range_scan <span class="version-mark">New in v5.0</span>

> **Note:**
>

> > Starting from v8.4.0, the default for this variable was changed from `OFF` to `ON`.

Reviewer: ```suggestion
> Starting from v8.4.0, the default value of this variable is changed from `OFF` to `ON`.
```

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): Yes
- Type: Boolean
- Default value: `ON`

> - After you set the value of this variable to `ON`, the optimizer will prefer range scans over full table scans for tables without statistics (pseudo statistics) or if the table is empty (zero statistics).

Reviewer 2: ```suggestion
- When the value of this variable is `ON`, the optimizer prefers range scans over full table scans for tables without statistics (pseudo statistics) or empty tables (zero statistics).
```
